lib/web/scripts/install.go

```diff
@@ -0,0 +1,158 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package scripts
+
+import (
+	"context"
+	_ "embed"
+	"strconv"
+	"strings"
+
+	"github.com/google/safetext/shsprintf"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
+)
+
+// AutoupdateStyle represents the kind of autoupdate mechanism the script should use.
+type AutoupdateStyle int
+
+const (
+	// NoAutoupdate means the installed Teleport should not autoupdate.
+	NoAutoupdate AutoupdateStyle = iota
+	// PackageManagerAutoupdate means the installed Teleport should update via a script triggering package manager
+	// updates. The script lives in the 'teleport-ent-update' package and was our original attempt at automatic updates.
+	PackageManagerAutoupdate
+	// UpdaterBinaryAutoupdate means the installed Teleport should update via the teleport-update binary.
+	// This update style does not depend on any package manager (although it has a system dependency to wake up the
+	// updater).
+	UpdaterBinaryAutoupdate
+)
+
+// InstallScriptOptions contains the Teleport installation options used to generate installation scripts.
+type InstallScriptOptions struct {
+	AutoupdateStyle AutoupdateStyle
+	// TeleportVersion that should be installed. Without the leading "v".
+	TeleportVersion string
+	// CDNBaseURL is the URL of the CDN hosting teleport tarballs.
+	// If left empty, the 'teleport-update' installer will pick the one to use.
+	CDNBaseURL string
+	// ProxyAddr is the address of the Teleport Proxy service that will be used
+	// by the updater to fetch the desired version. Teleport Addrs are
+	// 'hostname:port' (no scheme nor path).
+	ProxyAddr string
+	// TeleportFlavor is the name of the Teleport artifact fetched from the CDN.
+	// Common values are "teleport" and "teleport-ent".
+	TeleportFlavor string
+	// FIPS represents if the installed Teleport version should use Teleport
+	// binaries built for FIPS compliance.
+	FIPS bool
+}
+
+// Check validates that the minimal options are set.
+func (o *InstallScriptOptions) Check() error {
+	switch o.AutoupdateStyle {
+	case NoAutoupdate, PackageManagerAutoupdate:
+		return nil
+	case UpdaterBinaryAutoupdate:
+		// We'll do the checks later.
+	default:
+		return trace.BadParameter("unsupported autoupdate style: %v", o.AutoupdateStyle)
+	}
+	if o.ProxyAddr == "" {
+		return trace.BadParameter("Proxy address is required")
+	}
+
+	if o.TeleportVersion == "" {
+		return trace.BadParameter("Teleport version is required")
+	}
+
+	if o.TeleportFlavor == "" {
+		return trace.BadParameter("Teleport flavor is required")
+	}
+
+	if o.CDNBaseURL != "" && !strings.HasPrefix(o.CDNBaseURL, "https://") {
+		return trace.BadParameter("CDNBaseURL must start with 'https://'")
+	}
```

Comment:
Instead of checking this, can we instead try to parse as URL?
This can be loaded as user input, so we should be extra cautious here.